plonky3/src/prover/mod.rs

```diff
@@ -97,13 +98,68 @@ mod tests {
         assert!(proof.is_ok());
     }
 
+    fn run_test_goldilocks_publics(
```

Comment:
To avoid duplicated code, we should have `run_test_goldilocks` call `run_test_goldilocks_publics` with `None` as malicious inputs